include/onnxruntime/core/framework/ortdevice.h

```diff
@@ -62,6 +77,8 @@ struct OrtDevice {
     auto h = std::hash<int>()(device_type);
     onnxruntime::HashCombine(memory_type, h);
     onnxruntime::HashCombine(device_id, h);
+    if (alignment)
+      onnxruntime::HashCombine(*alignment, h);
```

Comment:
Would this be more deterministic?
```suggestion
    onnxruntime::HashCombine(alignment ? *alignment : 0, h);
```